File: NgCmsApi/Models/RouteCreateModel.cs
using System;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;
using System.Collections.Generic;

namespace NgCmsApi.Models
{
    public class RouteCreateModel
    {
        [Required]
        [JsonProperty("path")]
        public string Path { get; set; }

        [JsonProperty("parentRouteGuid")]
        public Guid? ParentRouteGuid { get; set; }
    }
}

File: NgCmsWeb/src/app/models/route-create.model.ts
import { Serializable } from './../interfaces/serializable.interface';

export class RouteCreateModel implements Serializable<RouteCreateModel> {
    path: string;
    parentRouteGuid: string;

    deserialize(input) {
        this.path = input.path;
        this.parentRouteGuid = input.parentRouteGuid;

        return this;
    }
}
